### PR TITLE
Change the default scaling for EUI to AsinhStretch

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/sunpy/sunpy/issues/6254 

Could also change EUVI to be the same?

TODO:
- [x] Check the changelog links without error

### CODE REVIEW

File: changelog/6285.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Changed the default scaling for `EUIMap` from a linear stretch to a asinh stretch.
```
Comment (nabobalis) [inline changelog/6285.bugfix.rst:1]:
```suggestion
Changed the default scaling for `~sunpy.map.sources.EUIMap` from a linear stretch to a asinh stretch.
```

**Comment by wtbarnes** [comment]
Why also EUVI?

**Comment by nabobalis** [comment]
> Why also EUVI?

It's just different. I have no actual reason.

**Comment by Cadair** [comment]
This doesn't feel like a thing we should backport?

**Comment by wtbarnes** [comment]
> > Why also EUVI?
> 
> It's just different. I have no actual reason.

image

The current scaling looks pretty good?

File: changelog/6285.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Changed the default scaling for `~sunpy.map.sources.EUIMap` from a linear stretch to a asinh stretch.
```
Comment (Cadair) [inline changelog/6285.bugfix.rst:1]: What's the shortest path to restoring the previous behaviour? Especially if we are backporting this we should tell people how to get back to the previous behaviour.

Comment (Cadair) [reply changelog/6285.bugfix.rst:1]:
```suggestion
Changed the default scaling for `~sunpy.map.sources.EUIMap` from a linear stretch to a asinh stretch.

To revert to the previous linear stretch do the following::

     from astropy.visualization import ImageNormalize, LinearStretch
     euimap.plot_settings["norm"] = ImageNormalize(stretch=LinearStretch())
```

**Comment by nabobalis** [comment]
> > > Why also EUVI?
> > 
> > 
> > It's just different. I have no actual reason.
> 
> image
> 
> The current scaling looks pretty good?

It could be better.

**Pull request merged**